gpm_api/tests/test_visualization/test_plot.py

```diff
@@ -0,0 +1,451 @@
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_cbar_kwargs(
+        self,
+        orbit_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with colorbar keyword arguments"""
+
+        cbar_kwargs = {"ticklabels": [42, 43, 44, 45, 46]}
+
+        plot.plot_map(orbit_dataarray, cbar_kwargs=cbar_kwargs)
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_rgb(
+        self,
+        orbit_rgb_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with RGB flag"""
+
+        plot.plot_map(orbit_rgb_dataarray, rgb=True)
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_rgb_invalid(
+        self,
+        orbit_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data with RGB flag on non RGB data"""
+
+        with pytest.raises(ValueError):
+            plot.plot_map(orbit_dataarray, rgb=True)
+
+    def test_grid(
+        self,
+        grid_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting grid data"""
+
+        plot.plot_map(grid_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_grid_antimeridian(
+        self,
+        grid_antimeridian_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting grid data going over the antimeridian"""
+
+        plot.plot_map(grid_antimeridian_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_grid_nan_lon(
+        self,
+        grid_nan_lon_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting grid data with NaN longitudes"""
+
+        plot.plot_map(grid_nan_lon_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_invalid(
+        self,
+    ) -> None:
+        """Test invalid data"""
+
+        da = xr.DataArray()
+        with pytest.raises(ValueError):
+            plot.plot_map(da)
+
+
+class TestPlotImage:
+    """Test the plot_image function"""
+
+    def test_orbit(
+        self,
+        orbit_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data"""
+
+        plot.plot_image(orbit_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_grid(
+        self,
+        grid_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting grid data"""
+
+        plot.plot_image(grid_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_invalid(
+        self,
+    ) -> None:
+        """Test invalid data"""
+
+        da = xr.DataArray()
+        with pytest.raises(ValueError):
+            plot.plot_image(da)
+
+
+class TestPlotMapMesh:
+    """Test the plot_map_mesh function"""
+
+    def test_orbit(
+        self,
+        orbit_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data"""
+
+        plot.plot_map_mesh(orbit_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_antimeridian(
```

Comment:
The bad results in the picture arise from the setting of the argument `edgecolors`, because the cell lines crossing the antimeridian (projection boundary) are not masked (only cell facecolors are masked). 
We could open an issue in cartopy to see if the maintainers foresee an easy solution to also mask the cell edges crossing the projection boundaries.

I don't see a straightforward solution in GPM-API for this case: 
- if the mesh to be plot is local (a small region crossing the antimeridian) we could detect when crossing the antimeridian only once and modify the CRS if is set to PlateeCarre projection and center it to 180 instead of 0. 
- But if the mesh to be plot is of multiple orbits around the earth, the only solution is to call the cartopy pcolormesh function only over orbit along-track slices where any cell intersects the antimeridian.

Reply:
I think we can archive this comment. 
Anything can't be done except a fix in cartopy. We can open a issue there in case